mailet/standard/src/main/java/org/apache/james/transport/mailets/StripAttachment.java

```diff
@@ -346,71 +388,99 @@ private boolean checkMessageRemoved(Part part, Mail mail)
+    }
+
+    private Optional<Boolean> isMatchingPattern(String name, Pattern pattern) {
+        if (pattern != null) {
+            return Optional.of(pattern.matcher(name).matches());
+        }
+        return Optional.absent();
+    }
+
     /**
-     * Saves the content of the part to a file in the given directoy, using the
-     * name of the part. If a file with that name already exists, it will
+     * Saves the content of the part to a file in the given directory, using the
+     * name of the part. Created files have unique names.
      * 
      * @param part
      *            The MIME part to save.
      * @return
      * @throws Exception
      */
-    private String saveAttachmentToFile(Part part, String fileName)
-            throws Exception {
-        BufferedOutputStream os = null;
-        InputStream is = null;
-        File f = null;
+    @VisibleForTesting Optional<String> saveAttachmentToFile(Part part, Optional<String> fileName) throws Exception {
         try {
-            if (fileName == null)
-                fileName = part.getFileName();
-            int pos = -1;
-            if (fileName != null) {
-                pos = fileName.lastIndexOf(".");
-            }
-            String prefix = pos > 0 ? (fileName.substring(0, pos)) : fileName;
-            String suffix = pos > 0 ? (fileName.substring(pos)) : ".bin";
-            while (prefix.length() < 3)
-                prefix += "_";
-            if (suffix.length() == 0)
-                suffix = ".bin";
-            f = File.createTempFile(prefix, suffix, new File(directoryName));
-            log("saving content of " + f.getName() + "...");
-            os = new BufferedOutputStream(new FileOutputStream(f));
-            is = part.getInputStream();
-            if (!(is instanceof BufferedInputStream)) {
-                is = new BufferedInputStream(is);
-            }
-            int c;
-            while ((c = is.read()) != -1) {
-                os.write(c);
-            }
+            File outputFile = outputFile(part, fileName);
+
+            log("saving content of " + outputFile.getName() + "...");
+            IOUtils.copy(part.getInputStream(), new FileOutputStream(outputFile));
 
-            return f.getName();
+            return Optional.of(outputFile.getName());
         } catch (Exception e) {
-            log("Error while saving contents of ["
-                    + (f != null ? f.getName() : (part != null ? part
-                            .getFileName() : "NULL")) + "].", e);
-            throw e;
-        } finally {
-            is.close();
-            os.close();
+            log("Error while saving contents of", e);
+            return Optional.absent();
+        }
+    }
+
+    private File outputFile(Part part, Optional<String> fileName) throws MessagingException, IOException {
+        if (!fileName.isPresent()) {
+            return createTempFile(part.getFileName());
         }
+        return createTempFile(fileName.get());
```

Comment:
if you use the bad isPresent/get() pattern, please use it right: if (filename.isPresent()) ... fileName.get() else ...

Or use return createTempFile(fileName.get().orElse(part.getFileName))